lib/azure/azure_interface.rb

```diff
@@ -0,0 +1,76 @@
+#
+# Author::
+# Copyright:: Copyright (c) 2016 Opscode, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'azure/custom_errors'
+
+class Azure
+  class AzureInterface

```

Comment:
There are a few deviations from the methods at https://github.com/chef/knife-cloud/blob/master/lib/chef/knife/cloud/service.rb -- can you give some background as to why some methods were left out or have a different signature?